torch/csrc/autograd/engine.cpp

```diff
@@ -850,19 +850,13 @@ void Engine::execute_graph_task_with_continuation(const std::shared_ptr<GraphTas
     // 'mark_graph_task_completed' would mark the Future as completed and this
     // would notify the owner thread that the task has been completed.
     mark_graph_task_completed(graph_task);
-  } else {
-    // schedule a continuation
-    at::launch([this, graph_task]() {
-        execute_graph_task_with_continuation(graph_task);
-    });
   }
-
+  return graph_task->future_result_;
```

Comment:
this PR doesn't address refcount issue? Without that fix, this will cause training failures.

Reply:
it seems that we don't need to change to return `shared_ptr<FutureVariableList>&` as of @jjlilley 's change in https://github.com/pytorch/pytorch/pull/36502, I will need to verify if this is the case.